Preprocessor guards for GL version-specific code

Peaches491 opened the pull request:
When attempting to run NanoGUI under GLES2, a number of GL3 functions were not available. This pull guards against those with preprocessor `#ifdef`s, and removes the hard-coded `#define NANOVG_GL3_IMPLEMENTATION` in `src/screen.cpp`

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/zooxco/nanogui/1)
<!-- Reviewable:end -->

Peaches491 committed:
Begin removal of GL3 specific code

Peaches491 committed:
Add ifdef guards around GL3-specific features

Peaches491 committed:
Add ifdef guards for cursors

Peaches491 committed:
Guard against glBindSampler

cgibson commented:
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/screen.cpp, line 35 at r1](https://reviewable.io:443/reviews/zooxco/nanogui/1#-KXqeuTQM0q1LmUumcyn:-KXqeuTQM0q1LmUumcyo:b-bk0o2e) ([raw file](https://github.com/zooxco/nanogui/blob/2f8498f52b4dc6dc54fd7c62d7e74ba7e470edea/src/screen.cpp#L35)):*
> ```cpp
> 
> /* Allow enforcing the GL2 implementation of NanoVG */
> //#define NANOVG_GL3_IMPLEMENTATION
> ```

We can probably get rid of this

---

*[src/screen.cpp, line 354 at r1](https://reviewable.io:443/reviews/zooxco/nanogui/1#-KXqf7SYBMpSCvDKAjGp:-KXqf7SYBMpSCvDKAjGq:b-693rar) ([raw file](https://github.com/zooxco/nanogui/blob/2f8498f52b4dc6dc54fd7c62d7e74ba7e470edea/src/screen.cpp#L354)):*
> ```cpp
> #ifndef NANOVG_CURSOR_DISABLED
>     for (int i=0; i < (int) Cursor::CursorCount; ++i)
>         mCursors[i] = glfwCreateStandardCursor(0x00036001 + i);
> ```

Any reason we changed to the value instead of GLFW_ARROW_CURSOR?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zooxco/nanogui/1)*
<!-- Sent from Reviewable.io -->

Peaches491 committed:
Renamed CURSOR macro, and updated ifdef style

Peaches491 committed:
More `#if defined` syntax fixes

Peaches491 commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/screen.cpp, line 35 at r1](https://reviewable.io:443/reviews/zooxco/nanogui/1#-KXqeuTQM0q1LmUumcyn:-KXr7eHf0G3SDboarqJA:b-896fix) ([raw file](https://github.com/zooxco/nanogui/blob/2f8498f52b4dc6dc54fd7c62d7e74ba7e470edea/src/screen.cpp#L35)):*
<details><summary><i>Previously, cgibson (Chris Gibson) wrote…</i></summary><blockquote>

We can probably get rid of this
</blockquote></details>

Done.

---

*[src/screen.cpp, line 354 at r1](https://reviewable.io:443/reviews/zooxco/nanogui/1#-KXqf7SYBMpSCvDKAjGp:-KXr7h1hyW71HY9ujhn0:b-b1mrv1) ([raw file](https://github.com/zooxco/nanogui/blob/2f8498f52b4dc6dc54fd7c62d7e74ba7e470edea/src/screen.cpp#L354)):*
<details><summary><i>Previously, cgibson (Chris Gibson) wrote…</i></summary><blockquote>

Any reason we changed to the value instead of GLFW_ARROW_CURSOR?
</blockquote></details>

Whoops! Reverted.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zooxco/nanogui/1)*
<!-- Sent from Reviewable.io -->

cgibson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zooxco/nanogui/1#-:-KXrCAUYMyP95hFwtavV:bnfp4nl)*
<!-- Sent from Reviewable.io -->
